FBKVOController/FBKVOController.m

```diff
@@ -268,6 +273,14 @@ - (void)observe:(id)object info:(_FBKVOInfo *)info
   
   // add observer
   [object addObserver:self forKeyPath:info->_keyPath options:info->_options context:(void *)info];
+
+  info->_registered = YES;
```

Comment:
You can actually get away here with just a single enum here, instead of 2 variables.
For example:
```objc
typedef NS_ENUM(uint8_t, _FBKVOInfoState) {
  _FBKVOInfoStateInitial = 0,
  _FBKVOInfoStateObserving,
  _FBKVOInfoStateNotObserving
}
```

And then you would simply remove the observer from the object in this very method if it is unobserved, otherwise you would set the state to observed.